utility.py

```diff
@@ -35,4 +35,31 @@ def distance(pos1: Tuple[float, float], pos2: Tuple[float, float]):
     """
     x1, y1 = pos1
     x2, y2 = pos2
-    return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
\ No newline at end of file
+    return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
+
+
+def position_on_edge(edge: sumolib.net.edge.Edge, pos: int):
+    """
+    :return: coordinate for pos meters from the start of the edge, following any shapes along edge
+    """
+    # Go through pair of coords, until meeting an edge, where if we travel through it, we have moved more than pos
+    # meters in total
+    remaining_distance = pos
+    for coord1, coord2 in (edge.getShape()[i:i + 2] for i in range(0, int(len(edge.getShape()) / 2 * 2), 2)):
```

Comment:
```suggestion
    for coord1, coord2 in (edge.getShape()[i:i + 2] for i in range(0, int(len(edge.getShape())), 2)):
```
That seems a bit unnecessary

Reply:
I have no idea what you are talking about

Reply:
Seems fixed in 796b834 :+1: